metaflow/runtime.py

```diff
@@ -826,16 +858,22 @@ def fds(self):
         return (self._proc.stderr.fileno(),
                 self._proc.stdout.fileno())
 
-    def kill(self):
-        if not self.killed:
+    def pre_kill(self):
```

Comment:
maybe `clean` instead of `pre_kill` and `self.cleaned` instead of `self.prekilled`?